src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -492,6 +500,14 @@ else if (restrictions.keyIsInRelation())
         try (PartitionIterator page = pager.fetchPage(pageSize, queryStartNanoTime))
         {
             msg = processResults(page, options, selectors, nowInSec, userLimit);
+
+            RequestSensors requestSensors = RequestTracker.instance.get();
+            if (requestSensors != null)
+            {
+                Optional<Sensor> readRequestSensor = RequestTracker.instance.get().getSensor(Type.READ_BYTES);
+                readRequestSensor.map(SensorsCustomParams::sensorValueAsByteBuffer)
+                                 .ifPresent(bytes -> msg.setCustomPayload(Collections.singletonMap(SensorsCustomParams.READ_BYTES_REQUEST, bytes)));
```

Comment:
> Note that the rate of BRg/BWg (or eventual RL) will be embedded in the custom payload on the CNDB side. Then, the customPayload should be merged to avoid overwriting this value.

See [SensorsRegistryAggregator ](https://github.com/datastax/cassandra/pull/952/commits/b4362cda6e1a941f7413c2ed556158bd19a88d72#diff-53f9479bcb5ca1193fee3a73e27649257714f50fe0c1897fc8da200bc7159e2a) for a proposal to add all custom payloads on C* side.

Haven't looked into the testing question yet.

Reply:
> Haven't looked into the testing question yet.

test proposal https://github.com/datastax/cassandra/pull/952/commits/a5ab081b805a9a714b1f94c372c966ecddf38532

Notes:
* The writer/coordinator running in the same process causes double counting on `SensorsRegistry` for the same tracked sensor type
* If it is ok to expose the `ResultMessage` on the `dtest-api` I can track those changes. Otherwise, in principle we could use native client to execute a message and parse the response for custom payload (something like https://github.com/datastax/cassandra/blob/0f25d1e31caa429fc2ebdb8574f5afe1c27d2f4b/test/distributed/org/apache/cassandra/distributed/test/UnableToParseClientMessageTest.java#L138) . However, I couldn't make run with distributed test nodes.